pkg/pool-weighted/test/foundry/SeedlessLBPPool.t.sol

```diff
@@ -0,0 +1,926 @@
+        }
+
+        // Check reserve accumulated
+        uint256[] memory midBalances = vault.getCurrentLiveBalances(pool);
+        assertApproxEqRel(midBalances[reserveIdx], totalBought, 0.01e18, "Reserve should accumulate from buys");
+
+        vm.warp(block.timestamp + DEFAULT_END_OFFSET + 1);
+
+        // Owner withdraws proceeds
+        uint256 bptBalance = IERC20(pool).balanceOf(bob);
+        uint256 reserveBefore = reserveToken.balanceOf(bob);
+        uint256 projectBefore = projectToken.balanceOf(bob);
+
+        uint256[] memory minAmountsOut = new uint256[](2);
+        vm.prank(bob);
+        router.removeLiquidityProportional(pool, bptBalance, minAmountsOut, false, bytes(""));
+
+        uint256 reserveReceived = reserveToken.balanceOf(bob) - reserveBefore;
+        uint256 projectReceived = projectToken.balanceOf(bob) - projectBefore;
+
+        // Owner should get all the accumulated reserve (proceeds) and remaining project tokens
+        assertApproxEqRel(reserveReceived, totalBought, 0.01e18, "Owner should receive sale proceeds");
+        assertGt(projectReceived, 0, "Owner should receive remaining project tokens");
+    }
+
+    /*******************************************************************************
+                                    PoolInfo Functions
+    *******************************************************************************/
+
+    function testGetCurrentLiveBalancesIncludesVirtual() public view {
+        uint256[] memory poolBalances = IPoolInfo(pool).getCurrentLiveBalances();
+        uint256[] memory vaultBalances = vault.getCurrentLiveBalances(pool);
+
+        assertEq(poolBalances[projectIdx], vaultBalances[projectIdx], "Project balance mismatch");
+        assertEq(
+            poolBalances[reserveIdx],
+            vaultBalances[reserveIdx] + reserveTokenVirtualBalance,
+            "Reserve should include virtual"
+        );
+    }
+
+    function testGetTokenInfoIncludesVirtual() public view {
+        (, , uint256[] memory balancesRaw, uint256[] memory lastBalancesLiveScaled18) = IPoolInfo(pool).getTokenInfo();
+
+        (, , uint256[] memory realBalancesRaw, uint256[] memory realLastBalancesLiveScaled18) = vault.getPoolTokenInfo(
+            pool
+        );
+
+        assertEq(balancesRaw[projectIdx], realBalancesRaw[projectIdx], "Project raw balance mismatch");
+        assertEq(
+            lastBalancesLiveScaled18[projectIdx],
+            realLastBalancesLiveScaled18[projectIdx],
+            "Project live balance mismatch"
+        );
+
+        assertEq(
+            balancesRaw[reserveIdx],
+            realBalancesRaw[reserveIdx] + reserveTokenVirtualBalance,
+            "Reserve raw should include virtual"
+        );
+        assertEq(
+            lastBalancesLiveScaled18[reserveIdx],
+            realLastBalancesLiveScaled18[reserveIdx] + reserveTokenVirtualBalance,
+            "Reserve live should include virtual"
+        );
```

Comment:
Might be worth doing the test with a 6-decimal reserve token, JIC. That will force us to see a few edge cases.

Reply:
Added a "non18" infrastructure (duplicated vs. generalized, to avoid breaking existing tests)